tests/wave_tank/nwt.mpirun=8.output

```diff
@@ -0,0 +1,11 @@
+Printing level set at cell center closest to point (4.2396, 0.52) 
+0.01	-0.03998369123
+0.02	-0.0399505819
+0.03	-0.03991531588
+0.04	-0.03988280904
+0.05	-0.03985496371
+0.06	-0.03983229365
+0.07	-0.03981465848
+0.08	-0.03980157022
+0.09	-0.03979253903
+0.1	-0.03978689494
```

Comment:
I get substantially different output for this test:
```
0.01    -0.03998478538
0.02    -0.03994721241
0.03    -0.0398957478
0.04    -0.03983655776
0.05    -0.0397740696
0.06    -0.03971252944
0.07    -0.03965461961
0.08    -0.03960119585
0.09    -0.03955297905
0.1     -0.03951051114
```
Do you know why this would be the case?

Reply:
I read through the code and I don't see anything like random number generation that could be giving different results on different platforms.

Reply:
The output file in tests/wave_tank/nwt.mpirun=8.output is run using: 
mpirun -np 4 ./nwt nwt.mpirun=8.input -stokes_ksp_rtol 1.0e-8

I have run it by using the following:
mpirun -np 8 ./nwt nwt.mpirun=8.input -stokes_ksp_rtol 1.0e-10

The output is (unable to upload the output file so pasting the output):

Printing level set at cell center closest to point (4.2396, 0.52)
0.01	        -0.03998368681
0.02	-0.03995058865
0.03	-0.03991532587
0.04	-0.0398828241
0.05	-0.03985500045
0.06	-0.03983235123
0.07	-0.0398147255
0.08	-0.03980168935
0.09	-0.03979270585
0.1	        -0.03978722421

This has a tighter tolerance than the earlier case.

Reply:
> The output file in tests/wave_tank/nwt.mpirun=8.output is run using:
> mpirun -np 4 ./nwt nwt.mpirun=8.input -stokes_ksp_rtol 1.0e-8
> 
> I have run it by using the following:
> mpirun -np 8 ./nwt nwt.mpirun=8.input -stokes_ksp_rtol 1.0e-10
> 
> The output is (unable to upload the output file so pasting the output):
> 
> Printing level set at cell center closest to point (4.2396, 0.52)
> 0.01 -0.03998368681
> 0.02 -0.03995058865
> 0.03 -0.03991532587
> 0.04 -0.0398828241
> 0.05 -0.03985500045
> 0.06 -0.03983235123
> 0.07 -0.0398147255
> 0.08 -0.03980168935
> 0.09 -0.03979270585
> 0.1 -0.03978722421
> 
> This has a tighter tolerance than the earlier case.

If you run the test (e.g. using `attest -R nwt`), does it pass for you?

Reply:
The test does not run. I have mentioned this in the earlier comments.

Reply:
> The test does not run. I have mentioned this in the earlier comments.

Ah, I misunderstood when you said that you had added a test case. For the test to be effective, the output in the repo needs to be the same as whatever is generated when the test is run by `attest`.

What goes wrong when you run your test?

Reply:
(Also, I would suggest not to check off "Does the test suite pass?" if the test suite is not working. 😁 )

Reply:
@drwells, if you use the output above, does the test suite work for you?

Reply:
> (Also, I would suggest not to check off "Does the test suite pass?" if the test suite is not working. grin )

I think that was me, actually, when we had all the tests passing a few days ago.

We will have to do some more debugging of the test runner but if `make tests; ./attest` doesn't work then that's beyond the scope of this PR. The right fix for this case (where we need an extra PETSc option) is to just set the option directly in the source file. Do you know how to do this?

Reply:
> We will have to do some more debugging of the test runner but if `make tests; ./attest` doesn't work then that's beyond the scope of this PR. The right fix for this case (where we need an extra PETSc option) is to just set the option directly in the source file. Do you know how to do this?

No. Could you show me how to do that please?

Reply:
Is the extra PETSc flag needed to get the right result?

Reply:
(I mean is the answer with lower tolerance “wrong” or simply less accurate?)

Reply:
You can add `PetscOptionsSetValue(nullptr, "-stokes_ksp_tol", "1e-10");` to the top of `main()` after you make the `IBTKInitialize` object (which initializes PETSc).

Reply:
> You can add `PetscOptionsSetValue(nullptr, "-stokes_ksp_tol", "1e-10");` to the top of `main()` after you make the `IBTKInitialize` object (which initializes PETSc).

`PetscOptionsSetValue(nullptr, "-stokes_ksp_rtol", "1e-10");`?

Reply:
Oh yeah, I forgot the r.

Reply:
> You can add `PetscOptionsSetValue(nullptr, "-stokes_ksp_rtol", "1e-10");` to the top of `main()` after you make the `IBTKInitialize` object (which initializes PETSc).

I added the above and the output is as follows:

Printing level set at cell center closest to point (4.2396, 0.52) 
0.01	        -0.03998368677
0.02	-0.03995058906
0.03	-0.03991532614
0.04	-0.03988282362
0.05	-0.0398549999
0.06	-0.03983235022
0.07	-0.03981472417
0.08	-0.03980168783
0.09	-0.03979270488
0.1	        -0.03978722355

Reply:
I discussed this with Amneet and he recommended to add the following lines in the nwt.mpirun=8.input file:

   // Solver parameters
   stokes_solver_type   = "VC_STAGGERED_STOKES_PETSC_KRYLOV_SOLVER"
   stokes_solver_db {
      options_prefix    = "stokes_"
      ksp_type          = "fgmres"
      rel_residual_tol  = 1e-14
      abs_residual_tol  = 1e-14
      max_iterations    = 1000
   }

This will make a tighter tolerance and should not create any difference in values in the output file.
I was also able to verify the test in one of the lab workstations using ./attest -R nwt and the test passed. Should I push these changes or do we need to investigate more regarding this issue?

Reply:
I agree — I think that putting these tolerances in the input file is better than hard coding the command line option in the source code.

Reply:
Oh yeah, that's a way better way to do it. Nice catch!

Reply:
In general, since tests are small enough to run quickly, its a good idea to use very tight tolerances so that we get consistent output even when our dependencies or algorithms change.